### ISSUE TITLE
Possible network problem causing high data usage

### ISSUE THREAD

**Issue by ozancaglayan** [DrKLO/Telegram#1089]
Hi,

I have a very annoying high data usage problem on my Android device: The "Android OS" keeps draining my cellular plan for some reason that I can't understand. I tried to debug the issue using tcpdump and interestingly got some pointers towards Telegram.

I captured the traffic for approx. 370 seconds and analyzed using wireshark. The phone was put aside, screen off. The only traffic on the capture was between me and 149.154.167.91:80 which I think is a telegram datacenter. The length of the TCP packets are all 0, they seem to be keepalive packets as far as I can understand. In 370 seconds, a total of 43506 packets have been transmitted and received which in total make a usage of 3000000 bytes of upload/download.

After a while,  I just got another burst of those packets and the mobile data usage increased again by ~2MB in android settings.

The interesting thing is that the process responsible of doing those transfers is shown as `system` in `netstat` output, not `org.telegram.messenger` which can explain high data usage of "Android OS":
```
tcp6       0     74 ::ffff:10.118.53.214:39380  ::ffff:149.154.167.91:80    FIN_WAIT1   system
```

I have everything disabled on my phone which can cause Android OS to consume mobile data: auto-sync, photo backup, android backup, etc. etc but I've got 668MB of background data usage from "Android OS" in 3 weeks.

A final observation is that the 668MB (so far) unknown data usage cumulated over 3 times actually, so it is not a continuous data burst. You can see the peaks from the screenshot below:

![index](https://cloud.githubusercontent.com/assets/330946/7672355/5f4c4b82-fcf8-11e4-8630-e0930fde6976.jpg)

Can you help me with this issue? I can also send you the `pcap` file of the burst if you want.
Thanks.

**Comment by ozancaglayan** [comment]
My operator is Vodafone, country is Turkey. I would also like to add that the data usage of "Android  OS" when wi-fi is on is about ~10MB for the same period compared against 668MB for 3G. This is also very strange but can be explained by the fact that I was always on 3G when this strange problem of data burst occurred.

**Comment by ly-nx** [comment]
Hi. What is the status?

Got similar issue for a while. Uninstalled the app for now and firewalled all others. 

![screenshot_2015-06-24-21-28-25](https://cloud.githubusercontent.com/assets/9435753/8338440/ce5dd89a-1ab9-11e5-9e2b-2928024890f7.png)

No spikes though.

![screenshot_2015-06-24-21-28-54](https://cloud.githubusercontent.com/assets/9435753/8338439/ce5be724-1ab9-11e5-89d3-b86bfda1f75c.png)

**Comment by ozancaglayan** [comment]
hello,

Nothing really new. I discussed with Markus from Telegram team and he sent me a debugged version of Telegram. I installed it on a friend's phone and mine, captured detailed logs and sent them to Markus but since then no response.

What is your phone and operator?

**Comment by ly-nx** [comment]
lg g2 / mts / 4.4.2 dorimanx kernel / cloudyfaG2 rooted. Not sure if there are many similar issues around.

**Comment by ly-nx** [comment]
![screenshot_2015-06-25-11-24-27](https://cloud.githubusercontent.com/assets/9435753/8350296/d47faf64-1b2c-11e5-824d-0ab9d81aeaae.png)

**Comment by ozancaglayan** [comment]
I hope to hear something from the developers but unfortunately nothing so far. There's also a discussion thread at http://androidforums.com/threads/holy-smokes-high-data-usage-by-android-os-solved.490265/ where I asked if anybody has Telegram and got 1 positive reply so far.

Annoying..

**Comment by ozancaglayan** [comment]
Although the data usage is shown as "Android OS", if your Android version allows, try disabling background data for Telegram and the data drain will stop.

**Comment by cevvalkoala** [comment]
I also noticed this "Android OS" process data leak and the battery drain it causes.

**Comment by ozancaglayan** [comment]
@cevvalkoala Can you also tell your phone model, android version and operator/country?

**Comment by cevvalkoala** [comment]
LG f3q, android 4.1.2, vodafone/turkey.

But disabling background data for telegram didn't stop the drain.

**Comment by ozancaglayan** [comment]
And switching to wireless, does it stop the issue?

**Comment by cevvalkoala** [comment]
Yes it does. No continuous data drain over wi-fi. Only on mobile data it bleeds.

**Comment by ozancaglayan** [comment]
I also reported the issue to Telegram through their support page:
https://telegram.org/support

**Comment by cevvalkoala** [comment]
It began on 23rd of June apparently. I don't know if it is a significant piece of information or not.
